index.ts

```diff
@@ -18,8 +18,48 @@ export class Tree<TValue extends number> {
   }
 
   private _insert(node: ITreeNode<TValue>, value: TValue) {
-    const direction = value >= node.value ? 'right' : 'left'
-    if (node[direction] !== undefined) this._insert(node[direction]!, value) // eslint-disable-line @typescript-eslint/no-non-null-assertion
+    const direction = this._setDirection(node, value)
+    /* eslint-disable */
+    if (node[direction] !== undefined) this._insert(node[direction]!, value)
     else node[direction] = { value }
+    /* eslint-enable */
+  }
+
+  private _setDirection(node: ITreeNode<TValue>, value: TValue) {
+    return value >= node.value ? 'right' : 'left'
+  }
+
+  public remove(value: TValue) {
+    if (this.root === undefined) throw new Error('There is no tree currently')
+    /* eslint-disable */
```

Comment:
eslint disable is not good

Reply:
![image](https://user-images.githubusercontent.com/64635043/172798751-3847ad06-ba62-4183-9610-ae03a3037151.png)

--> disable eslint for this purpose. 
--> `"@typescript-eslint/no-non-null-assertion": "off"` it is not recommended to turn off this rules for the entire code base